src/ng/compile.js

```diff
@@ -3526,6 +3570,11 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '&':
+            if (!hasOwnProperty.call(attrs, attrName) && !optional && strictComponentBindingsEnabled) {
+              throw $compileMinErr('missingattr',
```

Comment:
I would consider encapsulating this into a function or something, to avoid repeating it 4 times.